packages/app/src/files-manager/actions/upload-file.action.ts

```diff
@@ -16,11 +16,13 @@ import { getPathsRequest } from "App/file-system/requests"
 import { uploadFilesRequest } from "App/files-manager/requests"
 import { getFiles } from "App/files-manager/actions/get-files.action"
 import {
+  setPendingFilesToUpload,
   setUploadBlocked,
   setUploadingFileCount,
   setUploadingState,
 } from "App/files-manager/actions/base.action"
 import { loadStorageInfoAction } from "App/device/actions/load-storage-info.action"
+import { getFreeFilesSlotsCountForHarmony } from "App/files-manager/helpers/get-free-files-slots-count-for-harmony.helper"
 
 export const uploadFile = createAsyncThunk(
```

Comment:
Maybe we could move the logic by device type to standalone helpers or Mudita Harmony extended business logic? 🤔 

Option 1 - pseudo code
```suggestion
export const uploadFile = createAsyncThunk(
  FilesManagerEvent.UploadFiles,
    async (_, { getState, dispatch, rejectWithValue }) => {
      deviceType ? uploadFileViaMuditaHarmony() : uploadFileViaMuditaPure()
```

Option 2 - pseudo code
```suggestion
export const uploadFile = createAsyncThunk(
  FilesManagerEvent.UploadFiles,
    async (_, { getState, dispatch, rejectWithValue }) => {
      core...
      if(deviceType === MuditaHarmony) handleSpecificCaseForMuditaHarmony()
      return ...
```

Reply:
The topic has been discussed and will be resolved in the next iteration.